src/contextmenu.js

```diff
@@ -0,0 +1,472 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Multiple selection context menu.
+ */
+
+import * as Blockly from 'blockly/core';
+import {blockSelection, hasSelectedParent} from './global';
+
+/**
+ * Modification for context menu 'Duplicate' to be available for
+ * multiple blocks.
+ */
+const registerDuplicate = function() {
+  const duplicateOption = {
+    displayText: function() {
+      const selectedBlocksLength = blockSelection.size;
+      if (selectedBlocksLength <= 1) {
+        return Blockly.Msg['DUPLICATE_BLOCK'];
+      } else {
+        return Blockly.Msg['DUPLICATE_BLOCK'] + ' (' +
+        selectedBlocksLength + ')';
```

Comment:
This is a clever way to handle adding info while still using built-in translations!